### ISSUE TITLE
ItemAvailability: Comming, NotInStockYet

### ISSUE THREAD

**Issue by JendaPerl** [schemaorg/schemaorg#3472]
While there are three options for "you can't order this anymore" (and frankly I have no idea what's supposed to be the difference between Discontinued, OutOfStock and SoldOut) there is no option for "We are getting this ready, you can't order this YET, but you will be able to soon".
I mean, seriously, if you are selling, say, books, you do know the book is coming, you know when it is coming, you already know the price, but you can't start selling it before the official date. What are we supposed mark that?

**Comment by jvandriel** [comment]
Plus there's also the option to express when an `Offer` becomes available via [availabilityStarts](https://schema.org/availabilityStarts).

**Comment by kopojken** [comment]
https://schema.org/PreOrder:
"Indicates that the item is available for pre-order."

Not all coming products are available for pre-order. How about adding "notYetPublished" or "notYetOrderable"?

**Comment by jvandriel** [comment]
What I was trying to point out is that IMHO there's no need for a new enumeration to handle the situation you are describing @kopojken.

The following markup expresses the purpose of the `Offer` and when it becomes available, without having to specify any `itemAvailability`.
```
"offers": {
  "@type": "Offer",
  "url": "https://example.com/products/some-product/variant-1/,
  "businessFunction": "http://purl.org/goodrelations/v1#Sell",
  "itemCondition": "https://schema.org/NewCondition",
  "price": 49.95,
  "priceCurrency": "USD",
  "availabilityStarts": "2024-04-06T12:00:00+00:00"
}
```

**Comment by mfhepp** [comment]
You do not even need to attach a schema:Offer to a product entity that cannot/cannot yet be ordered. An order, in the original GoodRelations sense is the promise to transfer a bundle of rights (the business function) on something (a product, service, or bundle) for a compensation (price specification or other objects in the case of barter trade).

So when you are not yet ready to say if and when something can be ordered, you can either omit the order entity completely or leave it incomplete; e.g. by using a validFrom statement that is in the future and omitting all properties that are still unknown.

After all, we have a (pragmatic/diluted) open-world assumption in schema.org markup.

Before the flames come in re „but Google is nor showing xyz then“: That is something to be adressed with the consumers of the data; schema.org allows expressing a wide range of scenarios and a wide range of specificity.

**Comment by alex-jansen** [comment]
Closing this one, based on the discussion it seems already covered by the existing data model.
